### PR TITLE
Add train-test split functionality and more subsampling stubs

### PR DESCRIPTION
**Pull request by michaelkiper**
## Overview
This PR covers adding the train test split functionality for the processed static data, and the `subsample_from_config` stub.

As part of the static (raw) data processing, there will be a train-test split step (see below).
Screenshot 2025-09-02 at 1 15 52 PM

## Testing
This was done with unit tests in `tests/subsample/forward_pipeline_test.py`. Run `make test` to test, but it should be very straight forward.

### CODE REVIEW

File: src/cover_class/subsample/forward_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from typing import Tuple, Any
+from torch import FloatTensor, Tensor
+from numpy.typing import NDArray
+import numpy as np
+from sklearn.model_selection import train_test_split as tts # type: ignore[import]
+
+def train_test_split(data_matrix: FloatTensor, labels:Tensor, frac_test: float) -> Tuple[FloatTensor, FloatTensor, Tensor, Tensor]:
+    return tts(data_matrix, labels, test_size=frac_test)
+
```
Comment (pgbrodrick) [inline src/cover_class/subsample/forward_pipeline.py]: Can we default a random state in the function definition that gets passed through?  Unless we intentionally negate it, I'd like this reproducible.

Comment (marchett2) [reply src/cover_class/subsample/forward_pipeline.py]: --just to confirm -- the split is done per each class file. i.e. data_matrix and labels come from one pre-subsampled class? do we subsample to about an equal number of endmembers per class? trying to make sure that we do stratification for the split.

--we have an endmember library that has both ice and snow classes, same with ash and char. we combine these for the 6-class problem. is combining/merging labels done during the endmember subsampling?

Comment (michaelkiper) [reply src/cover_class/subsample/forward_pipeline.py]: @marchett2 those are good questions. For now, that functionality will not be included in this PR as this one if just to incorporate the ability to do the split, not execute it. I'll keep the stratified split in mind for where the part of the code that will execute it will be. 

In regards to the second question, any combined classes will be defined in the config level, and the subsampling will treat them as as a single class together. At least that's what I have in mind. However, that topic is out of scope for this PR. If you want to continue the conversation, please create an issue to track.

Comment (michaelkiper) [reply src/cover_class/subsample/forward_pipeline.py]: @pgbrodrick done

**Comment by michaelkiper** [comment]
@jakehlee tagging you in this PR too for a review (doesn't let me add more than 1 person since I'm on my free tier in a private repo 🫠)

**Review by jakehlee — APPROVED** [review]
LGTM, straightforward piping, more interested in how to do the stratified splitting.

File: src/cover_class/subsample/forward_pipeline.py (hunk 1 above)
Comment (pgbrodrick) [reply src/cover_class/subsample/forward_pipeline.py]: @marchett2 's points will get clear as we link pieces together.  There are a few ways to handle the combination, including on the input file side and the merges...I'm comfortable that this has the functionality we need, and that we just need to think through the, use, so will push forward to merge, but this is the right conversation.

**Review by pgbrodrick — APPROVED** [review]
